### PR TITLE
Update docs to DSL2 v2.0 syntax

### PR DESCRIPTION
**Pull request by jfy133**
Update main adding module guidelines/docs as per the new syntax

### CODE REVIEW

File: markdown/developers/adding_modules.md [hunk 1]
```diff
@@ -69,6 +71,8 @@ We have implemented a number of commands in the `nf-core/tools` package to make
 
 6. Create a module using the [nf-core DSL2 module template](https://github.com/nf-core/tools/blob/master/nf_core/module-template/modules/main.nf):
 
+    > ⚠️ [2021-11-26] please use `dev` version of tools
```
Comment (ggabernet) [inline markdown/developers/adding_modules.md:74]: This should be maybe also stated in step 1 above.

**Review by ggabernet — APPROVED** [review]
Looks good to me, just had a tiny comment

File: markdown/developers/adding_modules.md (hunk 1 above)
Comment (jfy133) [reply markdown/developers/adding_modules.md:78]: good point - done!

File: markdown/developers/adding_modules.md [hunk 2]
```diff
@@ -291,7 +299,18 @@ The key words "MUST", "MUST NOT", "SHOULD", etc. are to be interpreted as descri
...
+   ```nextflow
+   script:
+      def args = task.ext.args ?: ''
+      def prefix = task.ext.suffix ? "${meta.id}${task.ext.suffix}" : "${meta.id}"
+   """
+   fastqc \\
+     $args \\
+     <...>
+   """
+   ```
```
Comment (mahesh-panchal) [inline markdown/developers/adding_modules.md:313]:
```suggestion
    `<module>.nf`:
    ```nextflow
    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.suffix ? "${meta.id}${task.ext.suffix}" : "${meta.id}"
    """
    fastqc \\
        $args \\
         <...>
    """
    ```
    `modules.config`:
    ```nextflow
    process {
        withName: <module> {
            ext.args = [                                                                                                // Assign either a string, closure which returns a string
                '--quiet',
                params.fastqc_kmer_size ? "-k ${params.fastqc_kmer_size}" : ''     // parameter dependent values can be provided like so
            ].join(' ')                                                                                                    // join converts the list here to a string.
            ext.suffix = { "${meta.id}" }                                                     // A closure can be used to access variables defined in the script
        }
    }
    ```
```

File: markdown/developers/adding_modules.md [hunk 3]
```diff
@@ -460,20 +477,6 @@ using a combination of `bwa` and `samtools` to output a BAM file instead of a SA
 
 - If the software is not available on Bioconda a `Dockerfile` MUST be provided within the module directory. We will use GitHub Actions to auto-build the containers on the [GitHub Packages registry](https://github.com/features/packages).
 
-### Publishing results
```
Comment (mahesh-panchal) [inline markdown/developers/adding_modules.md:463]:
```suggestion
### Publishing results

Fomerly, results were published using a custom `publishDir` definition, customised using a Groovy Map defined by `params.modules`. This has been system has been replaced with using `publishDir`'s defined directly in a workflow's `modules.config` (see: ...)
```

**Review by mahesh-panchal — APPROVED** [review]
- Corrected that `ext.args` is a string, not a Map.
- I think an example of how it's used in the `modules.config` is useful.
- I think we should still have a publishing results section, particularly to explain to those who have familiarized themselves with the old system to see how it works now. 
- Do we need an explanation of how to modify the `nextflow.config` written here? It's purpose is only for testing right?

**Comment by jfy133** [comment]
> * Corrected that `ext.args` is a string, not a Map.
> * I think an example of how it's used in the `modules.config` is useful.
> * I think we should still have a publishing results section, particularly to explain to those who have familiarized themselves with the old system to see how it works now.
> * Do we need an explanation of how to modify the `nextflow.config` written here? It's purpose is only for testing right?

Good point with the publishing results, I'll select a good example from Harshil's he sent me last week.

for `nextflow.config`, yes we could do. I have an example in mind we could use (for testing, like you said))

**Comment by jfy133** [comment]
@mahesh-panchal have a look now, should've givne more examples

File: markdown/developers/adding_modules.md [hunk 4]
```diff
@@ -78,20 +83,20 @@ We have implemented a number of commands in the `nf-core/tools` package to make
...
                 ./modules/fastqc/main.nf
                 ./modules/fastqc/meta.yml
                 ./tests/modules/fastqc/main.nf
                 ./tests/modules/fastqc/test.yml
+                ./tests/modules/fastqc/nextflow.config
                 ./tests/config/pytest_modules.yml
     ```
 
-    All of the files required to add the module to `nf-core/modules` will be created/edited in the appropriate places. The 4 files you will need to change are:
+    All of the files required to add the module to `nf-core/modules` will be created/edited in the appropriate places. The 5 files you will need to or possibly change are:
```
Comment (mahesh-panchal) [inline markdown/developers/adding_modules.md:99]:
```suggestion
    All of the files required to add the module to `nf-core/modules` will be created/edited in the appropriate places. There are at most 5 files to modify:
```

File: markdown/developers/adding_modules.md [hunk 5]
```diff
@@ -107,7 +112,11 @@ We have implemented a number of commands in the `nf-core/tools` package to make
 
         Minimal test data required for your module may already exist within the [nf-core/modules repository](https://github.com/nf-core/modules/blob/master/tests/config/test_data.config), in which case you may just have to change a couple of paths in this file - see the [Test data](#test-data) section for more info and guidelines for adding new standardised data if required.
 
-    4. [`./tests/modules/fastqc/test.yml`](https://github.com/nf-core/modules/blob/master/tests/modules/fastqc/test.yml)
+    4. [`./tests/modules/fastqc/nextflow.config`](https://github.com/nf-core/modules/blob/master/tests/modules/amps/nextflow.config)
+
+        Some modules MAY require additional parameters added to the test command to successfully run. These can be specified with a `ext.args` variable within a process scope of the `nextflow.config`. file that exists alongside the test files themselves (and are automatically loaded when `main.nf` is executed.
```
Comment (mahesh-panchal) [inline markdown/developers/adding_modules.md:117]:
```suggestion
        Some modules MAY require additional parameters added to the test command to successfully run. These can be specified with an `ext.args` variable within the process scope of the `nextflow.config` file that exists alongside the test files themselves (and is automatically loaded when the test workflow `main.nf` is executed).
```

File: markdown/developers/adding_modules.md [hunk 6]
```diff
@@ -462,17 +496,7 @@ using a combination of `bwa` and `samtools` to output a BAM file instead of a SA
...
-    mode: params.publish_dir_mode,
-    saveAs: { filename -> saveFiles(filename:filename, options:params.options, publish_dir:getSoftwareName(task.process), publish_id:meta.id) }
-```
-
-The `saveFiles` function can be found in the [`functions.nf`](https://github.com/nf-core/modules/tree/master/modules/fastqc/functions.nf) file of utility functions that will be copied into all module directories. It uses the various publishing `options` specified as input to the module to construct and append the relevant output path to `params.outdir`.
-
-We also use a standardised parameter called `params.publish_dir_mode` that can be used to alter the file publishing method (default: `copy`).
+Fomerly, results were published using a custom `publishDir` definition, customised using a Groovy Map defined by `params.modules`. This has been system has been replaced with using Nextflow's native [`publishDir`](https://www.nextflow.io/docs/latest/process.html#publishdir) defined directly in a pipeline workflow's `modules.config` (see [here](https://github.com/nf-core/rnaseq/blob/f7702d5b76a1351e2e7796a5ed3f59943a139fbf/conf/modules.config#L100-L106) for a simple example)
```
Comment (mahesh-panchal) [inline markdown/developers/adding_modules.md:499]:
```suggestion
Fomerly, results were published using a custom `publishDir` definition, customised using a Groovy Map defined by `params.modules`. This system has been replaced using Nextflow's native [`publishDir`](https://www.nextflow.io/docs/latest/process.html#publishdir) defined directly in a pipeline workflow's `modules.config` (see [here](https://github.com/nf-core/rnaseq/blob/f7702d5b76a1351e2e7796a5ed3f59943a139fbf/conf/modules.config#L100-L106) for a simple example)
```

**Review by mahesh-panchal — APPROVED** [review]
Could the comments in the modules.config be aligned too please. It didn't work well for me.

**Comment by jfy133** [comment]
Done!

**Pull request merged**